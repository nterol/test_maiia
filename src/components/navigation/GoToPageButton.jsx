import React, { useCallback } from "react";


import { pageButton } from "../styles/navigation-button.module.scss";
import { goToPage } from "../../redux/actionTypes";
import { useDispatch } from "react-redux";
import { number, string } from "prop-types";

function GoToPageButton({ page, children }) {
  const dispatch = useDispatch();
  const dispatchSpecificPage = useCallback(
    (pageNumber) => dispatch({ type: goToPage, payload: pageNumber }),
    [dispatch],
  );
  return (
    <button data-testid="goto-page-button" className={`${pageButton} carved`} onClick={() => dispatchSpecificPage(page)}>
      {children || page}
    </button>
  );
}

GoToPageButton.defaultProps = {
  children: null,
};

GoToPageButton.propTypes = {
  page: number.isRequired,
  children: string,
};

export default GoToPageButton;
