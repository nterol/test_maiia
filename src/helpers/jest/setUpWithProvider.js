import React from "react";
import { render } from "@testing-library/react";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

const makeFakeStore = configureMockStore([]);

const setUpWithProvider = (Component, props = {}) => (mockStore) => {
  const fakeStore = makeFakeStore(mockStore);

  const tree = render(
    <Provider store={fakeStore}>
      <Component {...props} />
    </Provider>,
  );
  return tree;
};

export default setUpWithProvider;
