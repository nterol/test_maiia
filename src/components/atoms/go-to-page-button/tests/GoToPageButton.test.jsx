import setUpWithProvider from "../../../../helpers/jest/setUpWithProvider";

import GoToPageButton from "../GoToPageButton";

let setUpWithComponent;

test("GoToPage Button matches snapshot", () => {
  setUpWithComponent = setUpWithProvider(GoToPageButton, { page: 9 });
  const { container } = setUpWithComponent({});

  expect(container).toMatchSnapshot();
});

test("GoToPage Button matches snapshot with children", () => {
  setUpWithComponent = setUpWithProvider(GoToPageButton, {
    page: 9,
    children: "...",
  });
  const { getByTestId } = setUpWithComponent({});

  const button = getByTestId("goto-page-button");

  expect(button).toHaveTextContent("...");
});
