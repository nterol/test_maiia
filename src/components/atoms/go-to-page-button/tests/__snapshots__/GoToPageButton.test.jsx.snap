// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoToPage Button matches snapshot 1`] = `
<div>
  <button
    class="pageButton carved"
    data-testid="goto-page-button"
  >
    9
  </button>
</div>
`;
